proofs/clenv.ml

```diff
@@ -162,7 +165,9 @@ let clenv_environments evd bound t =
 let mk_clenv_from_env env sigma n (c,cty) =
   let evd = clear_metas sigma in
   let (evd,args,concl) = clenv_environments evd n cty in
-  { templval = mk_freelisted (mkApp (c,Array.map_of_list mkMeta args));
+  let templval = mkApp (c, Array.map_of_list mkMeta args) in
+  let metaset = Metaset.of_list args in
```

Comment:
I guess `c` is guaranteed to be meta free?

Reply:
Yes.